File: 06 - Seleção/pi_approximation.py
def pi(n):
    pi = 0

    for i in range(n):
        pi += (-1)**i * 4 / (2 * i + 1)
    return pi


print("Este programa calcula o valor aproximado de π fazendo uso da fórmula de Leibniz. O grau de precisão do resultado é dado pelo valor de epsilon, um número inteiro e não negativo.")

try:
    epsilon = int(input("Digite um valor para epsilon: "))
except ValueError:
    print("Erro: não é um valor inteiro.")
    exit()
except EOFError:
    print("")
    exit()
if epsilon < 0:
    print("Erro: valor negativo.")
    exit()
print("O valor aproximado de π para epsilon =", epsilon, "é", pi(epsilon))

File: 06 - Seleção/pi_approximation.c
#include <stdio.h>

int pow (int base, int exponent) {
	int result = 1;
	while (exponent-- > 0)
		result *= base;
	return result;
}

double pi (int n) {
	int i;
	double pi = 0;

	for (i = 0; i < n; i++)
		pi +=  pow(-1, i) * 4.0 / (2 * i + 1);
	return pi;
}

int main() {
	int epsilon;

	printf("Este programa calcula o valor aproximado de π fazendo uso da fórmula de Leibniz. O grau de precisão do resultado é dado pelo valor de epsilon, um número inteiro e não negativo.\nDigite um valor para epsilon: ");
	if (scanf(" %d", &epsilon) == EOF || epsilon < 0) {
		printf("Valor inválido");
		return 1;
	}
	printf("O valor aproximado de π para epsilon = %d é: %lf\n", epsilon, pi(epsilon));
	return 0;
}
